posydon/binary_evol/DT/double_CO.py

```diff
@@ -104,8 +104,9 @@ def ev_contact(t, y):
             t_final = [t_inspiral]
 
         if s.status == -1:
-            binary.state += ' (Integration failure)'
-            raise NumericalError("Integration failed: " + s.message)
+            failed_state = binary.state
+            set_binary_to_failed(binary)
+            raise NumericalError(f"Integration failed for {failed_state} DCO!")
```

Comment:
Why don't we keep the message from the solver added to the error message?

Similarly, it would be nice to add more information to all the errors, especially stuff which is otherwise not preserved in the binary object, e.g. `primary.co`.